internal/slicer/slicer_test.go

```diff
@@ -781,16 +894,183 @@ var slicerTests = []slicerTest{{
 			slices:
 				myslice:
 					contents:
-						/dir/nested/file:
+						/file:
 		`,
 	},
 	filesystem: map[string]string{
-		"/dir/":            "dir 0755",
-		"/dir/nested/":     "dir 0755",
-		"/dir/nested/file": "file 0644 84237a05",
+		// /file comes from the test-package in archive "bar" as the archive
+		// name is pinned in the slice definition file.
+		"/file": "file 0644 fa0c9cdb",
+	},
+	report: map[string]string{
+		"/file": "file 0644 fa0c9cdb {test-package_myslice}",
+	},
+}, {
+	summary: "Pinned archive does not have the package",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
+	archives: map[string]*testArchive{
+		"foo": {
+			pkgs: map[string][]byte{
+				"test-package": testutil.MustMakeDeb([]testutil.TarEntry{
+					testutil.Reg(0644, "./file", "from foo"),
+				}),
+			},
+		},
+		"bar": {
+			pkgs: map[string][]byte{},
+		},
+	},
+	release: map[string]string{
+		"chisel.yaml": `
+			format: chisel-v1
+			archives:
+				foo:
+					version: 22.04
+					components: [main, universe]
+					default: true
+					priority: 20
+					v1-public-keys: [test-key]
+				bar:
+					version: 22.04
+					components: [main]
+					priority: 10
+					v1-public-keys: [test-key]
+			v1-public-keys:
+				test-key:
+					id: ` + testKey.ID + `
+					armor: |` + "\n" + testutil.PrefixEachLine(testKey.PubKeyArmor, "\t\t\t\t\t\t") + `
+		`,
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			archive: bar
+			slices:
+				myslice:
+					contents:
+						/file:
+		`,
+	},
+	// Although archive "foo" does have the package, since archive "bar" has
+	// been pinned in the slice definition, no other archives will be checked.
+	error: `slice package "test-package" missing from archive`,
+}, {
+	summary: "No archives have the package",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
+	archives: map[string]*testArchive{
+		"foo": {
+			pkgs: map[string][]byte{},
+		},
+		"bar": {
+			pkgs: map[string][]byte{},
+		},
+	},
+	release: map[string]string{
+		"chisel.yaml": `
+			format: chisel-v1
+			archives:
+				foo:
+					version: 22.04
+					components: [main, universe]
+					default: true
+					priority: 20
+					v1-public-keys: [test-key]
+				bar:
+					version: 22.04
+					components: [main]
+					priority: 10
+					v1-public-keys: [test-key]
+			v1-public-keys:
+				test-key:
+					id: ` + testKey.ID + `
+					armor: |` + "\n" + testutil.PrefixEachLine(testKey.PubKeyArmor, "\t\t\t\t\t\t") + `
+		`,
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/file:
+		`,
+	},
+	error: "slice package \"test-package\" missing from archive\\(s\\)",
+}, {
+	summary: "Negative priority archives are ignored, unless asked by package",
```

Comment:
What about `Negative priority archives are ignore when not explicitly pinned in package`?